collect_app/src/main/java/org/odk/collect/android/http/CaseInsensitiveHeaders.java

```diff
@@ -0,0 +1,15 @@
+package org.odk.collect.android.http;
+
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public interface CaseInsensitiveHeaders {
+    @Nullable Set<String> getHeaders();
+
+    boolean containsHeader(String header);
+
+    @Nullable String getAnyValue(String header);
```

Comment:
The naming choice of 'Any' allows for the implementation to have its own approach for rapidly looking up a single value. Most client calls will only care about the case of a single value. If having multiple values is a concern, the client is responsible for navigating the whole collection.